library/src/utils/isSchema.ts

```diff
@@ -0,0 +1,12 @@
+import type { BaseSchema, BaseSchemaAsync } from '../types/schema.ts';
+import { isNonNullable } from './isNonNullable.ts';
+
+export const isSchema = (val: unknown): val is BaseSchema | BaseSchemaAsync =>
```

Comment:
Why do we need this util?

Reply:
I removed this when adding `hasType`. Strictly speaking I think this was a more thorough type guard for runtime validation that an object extends a Schema type, but it's not entirely necessary.